components/registry/org.wso2.carbon.registry.common.ui/src/main/resources/web/registry_common/js/registry_validation.js

```diff
@@ -188,3 +188,111 @@ function validateEmail(fld) {
     }
     return error;
 }
+
+function validateSimpleSearch() {
+    // JS injection validation
+    if (!validateTextForIllegal(document.forms["searchForm"]["criteria"], "resource path")) {
+        CARBON.showWarningDialog(org_wso2_carbon_registry_common_ui_jsi18n["the"] +
+        " " + "search content" + " " + org_wso2_carbon_registry_common_ui_jsi18n["contains.illegal.chars"]);
+        return false;
+    }
+
+    var searchText = document.forms["searchForm"]["criteria"].value;
+    if (searchText == null || searchText == "") {
+        CARBON.showWarningDialog(org_wso2_carbon_registry_search_ui_jsi18n["validate.simple.search"]);
+        return false;
+    }
+    document.forms['searchForm'].submit();
+    return true;
+}
+
+function validateFilterName(fld, fldName) {
+
+    var illegalChars = /\//; // do not allow slash
+    var fnReason = "";
+
+    fnReason += validateEmpty(fld, fldName);
+
+    if (fnReason == "") {
+        fnReason += validateForInput(fld, fldName);
+    }
+    if (fnReason == "") {
+        fnReason += validateIllegal(fld, fldName);
+    }
+    if (fnReason == "" && illegalChars.test(fld.value)) {
+        fnReason += org_wso2_carbon_registry_search_ui_jsi18n["filter.name.cannot.contain.slash"];
+    }
+
+    return fnReason;
+}
+
+function validatePropertyValues() {
+    var leftVal = parseInt(document.getElementById('valueLeft').value);
+    var rightVal = parseInt(document.getElementById('valueRight').value);
+
+    if (leftVal != "" && rightVal != "") {
+        if (leftVal >= rightVal) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+function validateEmptyPropertyValues() {
+
+    var leftVal = document.getElementById('valueLeft').value;
+    var rightVal = document.getElementById('valueRight').value;
+    var opRight = document.getElementById('opRight');
+    var propertyName = document.getElementById('#_propertyName').value;
+
+    if (leftVal != "" || rightVal != "") {
+        if (propertyName == "") {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+function validateIllegalContentSearchString(fld, fldName) {
+    var error = "";
+    //var illegalChars = /([^a-zA-Z0-9_\-\x2E\&\?\/\:\,\s\(\)\[\]])/;
+    var illegalChars = /([~!@#$;%^*+{}\|\\<>\"\',\[\]\(\)])/; // disallow ~!@#$;%^*+={}|\<>"',[]()
+    var illegalCharsInput = /(\<[a-zA-Z0-9\s\/]*>)/;
+    if (illegalChars.test(fld.value) || illegalCharsInput.test(fld.value)) {
+        error = org_wso2_carbon_registry_search_ui_jsi18n["the"] +
+        " " + fldName + " " + org_wso2_carbon_registry_search_ui_jsi18n["contains.illegal.chars"] + "<br />";
+    } else {
+//        fld.style.background = 'White';
```

Comment:
If there's no logic for else statement, why need one in the first place? Also remove commented lines